prime-dotnet-webapi/Controllers/SubmissionsController.cs

```diff
@@ -362,6 +362,45 @@ public async Task<ActionResult> AssignToaAgreementType(int enrolleeId, [FromBody
             return Ok(updatedEnrollee);
         }
 
+        // PUT: api/enrollees/5/status-actions/change-toa
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="enrolleeId"></param>
+        /// <param name="data"></param>
+        [HttpPut("{enrolleeId}/status-actions/change-toa", Name = nameof(ChangeToaAgreementType))]
+        [Authorize(Roles = Roles.ApproveEnrollee)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status404NotFound)]
+        [ProducesResponseType(typeof(ApiResultResponse<EnrolleeViewModel>), StatusCodes.Status200OK)]
+        public async Task<ActionResult> ChangeToaAgreementType(int enrolleeId, ChangeToaUpdateViewModel data)
+        {
+            if (!await _enrolleeService.EnrolleeExistsAsync(enrolleeId))
+            {
+                return NotFound($"Enrollee not found with id {enrolleeId}");
+            }
+
+            var assignedToaType = (data.AgreementType == 0) ? null : (AgreementType?)data.AgreementType;
+
+            if (assignedToaType.HasValue && !Enum.IsDefined(typeof(AgreementType), data.AgreementType))
+            {
+                return NotFound($"Agreement type not found with id {data.AgreementType}.");
+            }
+
+            if (assignedToaType.HasValue && !data.AgreementType.IsEnrolleeAgreement())
+            {
+                return BadRequest("Agreement type must be a TOA.");
+            }
+
+            if (!await _enrolleeService.IsEnrolleeInStatusAsync(enrolleeId, StatusType.Editable))
+            {
+                return BadRequest("Assigned agreement type can not be updated when the current status is not 'Under Review'.");
```

Comment:
Error message needs re-phrasing:  "Assigned agreement type may be updated only when the current status is Editable"

Reply:
Good catch. Updated.